### ISSUE TITLE
No gradient clipping  agent experiments can silently diverge

### ISSUE THREAD

**Issue by NJX-njx** [karpathy/autoresearch#8]
## Problem

The training loop has no gradient norm clipping. The only stability guard is a hard exit when \	rain_loss > 100\:

\\\python
if train_loss_f > 100:
    print(FAIL)
    exit(1)
\\\

When the autonomous agent experiments with aggressive hyperparameters (large learning rates, unusual architectures), gradient spikes can cause the loss to jump past the threshold in a single step, wasting the full 5-minute experiment. Gradient clipping would allow the training to recover from transient spikes instead of immediately diverging.

## Proposed Fix

Add an optional gradient norm clipping step before \optimizer.step()\, controlled by a hyperparameter:

\\\python
GRAD_CLIP_NORM = 1.0  # max gradient norm (0.0 = disabled)
\\\

This is a standard practice (used in the original nanochat) that adds one line to the training loop and one hyperparameter constant. Disabled by default if set to 0.0.

I'll submit a PR for this.
